packages/flutter/test/material/theme_data_test.dart

```diff
@@ -170,4 +170,138 @@ void main() {
+    expect(themeDataCopy.disabledColor, equals(darkTheme.disabledColor));
+    expect(themeDataCopy.buttonTheme, equals(darkTheme.buttonTheme));
+    expect(themeDataCopy.toggleButtonsTheme, equals(darkTheme.toggleButtonsTheme));
+    expect(themeDataCopy.buttonColor, equals(darkTheme.buttonColor));
+    expect(themeDataCopy.secondaryHeaderColor, equals(darkTheme.secondaryHeaderColor));
+    expect(themeDataCopy.textSelectionColor, equals(darkTheme.textSelectionColor));
+    expect(themeDataCopy.cursorColor, equals(darkTheme.cursorColor));
+    expect(themeDataCopy.textSelectionColor, equals(darkTheme.textSelectionColor));
+    expect(themeDataCopy.cursorColor, equals(darkTheme.cursorColor));
+    expect(themeDataCopy.textSelectionHandleColor, equals(darkTheme.textSelectionHandleColor));
+    expect(themeDataCopy.backgroundColor, equals(darkTheme.backgroundColor));
+    expect(themeDataCopy.dialogBackgroundColor, equals(darkTheme.dialogBackgroundColor));
+    expect(themeDataCopy.indicatorColor, equals(darkTheme.indicatorColor));
+    expect(themeDataCopy.hintColor, equals(darkTheme.hintColor));
+    expect(themeDataCopy.errorColor, equals(darkTheme.errorColor));
+    expect(themeDataCopy.toggleableActiveColor, equals(darkTheme.accentColor));
+    expect(themeDataCopy.textTheme, equals(darkTheme.textTheme));
+    expect(themeDataCopy.primaryTextTheme, equals(darkTheme.primaryTextTheme));
+    expect(themeDataCopy.accentTextTheme, equals(darkTheme.accentTextTheme));
+    expect(themeDataCopy.sliderTheme, equals(darkTheme.sliderTheme));
+    expect(themeDataCopy.tabBarTheme, equals(darkTheme.tabBarTheme));
+    expect(themeDataCopy.tooltipTheme, equals(darkTheme.tooltipTheme));
+    expect(themeDataCopy.cardTheme, equals(darkTheme.cardTheme));
+    expect(themeDataCopy.chipTheme, equals(darkTheme.chipTheme));
+    expect(themeDataCopy.platform, equals(darkTheme.platform));
+    expect(themeDataCopy.materialTapTargetSize, equals(darkTheme.materialTapTargetSize));
+    expect(themeDataCopy.applyElevationOverlayColor, equals(darkTheme.applyElevationOverlayColor));
+    expect(themeDataCopy.pageTransitionsTheme, equals(darkTheme.pageTransitionsTheme));
+    expect(themeDataCopy.appBarTheme, equals(darkTheme.appBarTheme));
+    expect(themeDataCopy.bottomAppBarTheme, equals(darkTheme.bottomAppBarTheme));
+    expect(themeDataCopy.colorScheme, equals(darkTheme.colorScheme));
+    expect(themeDataCopy.dialogTheme, equals(darkTheme.dialogTheme));
+    expect(themeDataCopy.floatingActionButtonTheme, equals(darkTheme.floatingActionButtonTheme));
+    expect(themeDataCopy.typography, equals(darkTheme.typography));
+    expect(themeDataCopy.cupertinoOverrideTheme, equals(darkTheme.cupertinoOverrideTheme));
+    expect(themeDataCopy.snackBarTheme, equals(darkTheme.snackBarTheme));
+    expect(themeDataCopy.bottomSheetTheme, equals(darkTheme.bottomSheetTheme));
+    expect(themeDataCopy.popupMenuTheme, equals(darkTheme.popupMenuTheme));
```

Comment:
This unexpectedly passed. I think you might, unfortunately, have to use custom `ThemeData` values instead of light vs dark theme, since some of these properties are identical in both themes.

Reply:
Right ! Now it's 2 distincts `ThemeData.raw` instances.